jerry-ext/args/args-internal.h

```diff
@@ -0,0 +1,31 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_ARGS_INTERNAL_H
+#define JERRYX_ARGS_INTERNAL_H
+
+#include "jerryscript.h"
+
+/**
+ * The iterator structor for JS arguments.
+ */
+struct jerryx_arg_js_iterator_t
```

Comment:
((Sure. The question is whether we want to allow them that? Whether we want to have a truly opaque iterator type or one that is only opaque to the users of the library? Even in the current for of the PR, arg-transform-functions.c includes args-internal.h but does not use it (all peeks and pops are performed with the help of helpers). Other than arg-js-iterator-helper.c, only args.c uses the knowledge of the internals of the iterator type, when jerryx_arg_transform_args_with_this and jerryx_arg_transform_args construct iterators. If we had one more helper -- say, jerryx_arg_js_iterator_create,construct,init --, then not even args.c had to know how the iterator type looks like.))

Scratch all that. That's not true. Because jerryx_arg_transform_args_with_this and jerryx_arg_transform_args have the iterator in a local variable (not just a pointer to an iterator), they have to know the size of the type, so they have to see the full declaration.

We do need the internal header. (But we don't necessarily have to include it in arg-transform-functions.c as it's unused there, that still holds.)